stdlib/public/Concurrency/AsyncStream.swift

```diff
@@ -167,7 +265,39 @@ public struct AsyncStream<Element> {
     build(Continuation(storage: storage))
   }
 
-  
+  /// Constructs an asynchronous stream from a given element-producing
+  /// closure, with an optional closure to handle cancellation.
+  ///
+  /// - Parameters:
+  ///   - produce: A closure that asynchronously produces elements for the
+  ///    stream.
+  ///   - onCancel: A closure to execute when cancelling the stream's task.
+  ///
+  /// Use this convenience initializer when you have an asychronous function
+  /// that can produce elements for the stream, and don't want to invoke
+  /// a continuation manually. This initializer "unfolds" your closure into
+  /// a full-blown asynchronous stream. The created stream handles adherence to
```

Comment:
This is the first time I've seen "adherence" to a protocol.  What the sentence is describing isn't quite protocol conformance, which is the usual term, but seems a bit more general.

Reply:
This is kind of borrowing from the [swift-evolution proposal](https://github.com/apple/swift-evolution/blob/main/proposals/0314-async-stream.md):

> This affords the case in which that unfolded function can leave the specification adherence of AsyncSequence to AsyncStream. In short the `init(unfolding:onCancel:)` initializer handles the terminal cases as well as the cancellation.

Reply:
Conformance might be better? Since it is a behavioral requirement of AsyncSequence (not a compile time enforced property)?

Or perhaps qualify it as "well behaved conformance"?

Reply:
I think we can just call it "conformance".  By analogy, if I made a type that implemented the protocol requirements of `Collection` but crashed if you tried to iterate more than 3 times, that type doesn't correctly conform to the `Collection` protocol.  Except in the context of discussing this sort of incorrect conformance, we shouldn't need to distinguish well-behaved conformance.

Reply:
> I think we can just call it "conformance".
> 
> By analogy, supposed I made a type that implemented the methods that `Collection` requires, but crashed if you tried to iterate more than 3 times. That type doesn't correctly conform to the `Collection` protocol because a collection must support multiple iteration. Many protocols have requirements that can't be expressed in code like this.
> 
> Except in the context of discussing this sort of incorrect conformance, we shouldn't need to distinguish well-behaved conformance.

@natecook1000 what do you think?

Reply:
Personally, I'm fine with "conformance". I kept "adherence" as a synonym borrowed from the API proposal, because `AsyncStream` conforms to `AsyncSequence` in the first place, so it's almost a tautology to talk about conformance at all. I think the real point here is that the developer sends in a closure and this initializer provides all the other piping to serve as an `AsyncStream`, which in turn is an `AsyncSequence`, which is what the final sentence is about.